projectJava/FilterWords/src/Filter.java

```diff
@@ -0,0 +1,170 @@
+import java.sql.SQLOutput;
+import java.util.Scanner;
+
+import static sun.nio.ch.IOStatus.check;
+
+public class Filter {
+    public static void main(String[] args){
+        Scanner sc = new Scanner(System.in);
+        String[] words;
+        boolean end = false;
+        do {
+            System.out.println("Пожалуйста, введите слова через запятую, минимальное количество слов - 3.");
+            String enteredWords = sc.nextLine();
+            words = enteredWords.split(", ");
+            if (!enteredWords.contains(", ") || words.length < 3) {
+                System.out.println("Неверное количество слов, попробуйте снова. ");
+            }else
+            {
+                end = true;
+            }
+        }while (!end);
+
+        boolean shutoff = false;
+        do{
+            System.out.println("Пожалуйста, выберите фильтр: ");
+            System.out.println("1 - выбрать слова с длиной меньше либо равно введенному значению");
+            System.out.println("2 - выбрать слова, начинающиеся с выбранной буквы");
+            System.out.println("3 - выбрать слова, заканчивающиеся на выбранную букву");
+            System.out.println("4 - выбрать слова, которые в себе содержат выбранную последовательность букв");
+            System.out.println("0 - выйти из программы");
+            System.out.println("Введите номер выбранного фильтра");
+            int filter = sc.nextInt();
+            boolean success = false ;
+
+            switch (filter) {
+                case 1:
+                    do {
+                        System.out.println("Введите сколько букв должно содержать слово: ");
+                        int length = sc.nextInt();
+
+                        String[] findedWords = new String[words.length];
+                        for (int i = 0; i < words.length; i++) {
+                            if (words[i].length() <= length) {
+                                findedWords[i] = words[i];
+                            }
+                        }
+                        String allFindedWords = "";
```

Comment:
Code from 47-65 is duplicated many times in different parts. It could be extracted to separate method (as IDEA suggest)
![image](https://user-images.githubusercontent.com/109154988/184238373-f2f46ac1-82b5-4cce-99df-cfcce0637554.png)